

package mm;

//内部公用错误协议
message ErrorMsg
{
    required int32 errcode = 1;
}

// 登录协议(服务器内部)
message LoginReq
{
    required uint64 userid = 1;
    required string passwd = 2;
}

message LoginRsp
{
    required uint64 userid = 1;
    required int32 ret = 2; // 1表示成功
}

// GNS登记协议
message GNSRegisterReq
{
    required uint64 userid = 1;
    required int32 serverid = 2;
    required uint32 connpos = 3;
}

message GNSRegisterRsp
{
    required uint64 userid = 1;
    required int32 serverid = 2;
    required uint32 connpos = 3;
    required int32 ret = 4; // 1表示成功
}

// GNS取消登记协议
message GNSUnRegisterReq
{
    required uint64 userid = 1;
    required int32 serverid = 2;
    required uint32 connpos = 3;
}

message GNSUnRegisterRsp
{
    required uint64 userid = 1;
    required int32 serverid = 2;
    required uint32 connpos = 3;
    required int32 ret = 4; // 1表示成功
}

// 通知连接层断开某条链接
message DisconnectReq
{
    required uint64 userid = 1;
    required int32 serverid = 2;
    required uint32 connpos = 3;
}

message DisconnectRsp
{
    required uint64 userid = 1;
    required int32 serverid = 2;
    required uint32 connpos = 3;
    required int32 ret = 4; // 1表示成功
}